x-pack/plugins/viz_editor/common/select_builder.ts

```diff
@@ -22,27 +22,38 @@ const operationTemplate: OperationTemplate = {
       interval: 'd',
     },
   },
+  window: {
+    operator: 'window',
+    argument: {
+      field: '',
+      windowFunction: 'unweightedAvg',
+      windowSize: 5,
+    },
+  },
 };
 
 export function getOperationTemplate(operator: SelectOperator) {
-  const template: SelectOperation = { ...operationTemplate[operator] };
+  const template: Partial<SelectOperation> = { ...operationTemplate[operator] };
 
   if ('argument' in template) {
     // Deep clone
-    template.argument = { ...template.argument };
+    template.argument = { ...template.argument } as any;
   }
 
   return template;
 }
 
 export function fieldToOperation(field: DatasourceField, operator: SelectOperator) {
-  const template: SelectOperation = getOperationTemplate(operator);
+  const template = {
+    ...getOperationTemplate(operator),
+    alias: field.name,
+  } as SelectOperation;
 
   if ('argument' in template) {
     template.argument.field = field.name;
   }
 
-  template.alias = field.name;
+  template.alias = operator === 'count' ? 'count' : field.name;
```

Comment:
How about `fieldToOperation` takes different arguments:

```
export function fieldToOperation(id: number, field: DatasourceField, operator: SelectOperator) {
```

Reply:
Makes sense, added. In the long run we should think about how to generate/maintain these ids in general. The current approach isn't perfect as the operations can get moved around/changed and the id won't reflect what's in them any more. Maybe we should just pick short (to avoid getting the expression too noisy) random strings